src/legacy/core_plugins/data/public/search/aggs/agg_configs.test.ts

```diff
@@ -0,0 +1,503 @@
+  });
+
+  describe('#getResponseAggs', () => {
+    it('returns all request aggs for basic aggs', () => {
+      const configStates = [
+        { type: 'terms', enabled: true, params: {}, schema: 'split' },
+        { type: 'date_histogram', enabled: true, params: {}, schema: 'segment' },
+        { type: 'count', enabled: true, params: {}, schema: 'metric' },
+      ];
+
+      const ac = new AggConfigs(indexPattern, configStates, { typesRegistry });
+      const sorted = ac.getResponseAggs();
+      const aggs = indexBy(ac.aggs, agg => agg.type.name);
+
+      expect(sorted.shift()).toBe(aggs.terms);
+      expect(sorted.shift()).toBe(aggs.date_histogram);
+      expect(sorted.shift()).toBe(aggs.count);
+      expect(sorted).toHaveLength(0);
+    });
+
+    it('expands aggs that have multiple responses', () => {
+      const configStates = [
+        { type: 'terms', enabled: true, params: {}, schema: 'split' },
+        { type: 'date_histogram', enabled: true, params: {}, schema: 'segment' },
+        { type: 'percentiles', enabled: true, params: { percents: [1, 2, 3] }, schema: 'metric' },
+      ];
+
+      const ac = new AggConfigs(indexPattern, configStates, { typesRegistry });
+      const sorted = ac.getResponseAggs();
+      const aggs = indexBy(ac.aggs, agg => agg.type.name);
+
+      expect(sorted.shift()).toBe(aggs.terms);
+      expect(sorted.shift()).toBe(aggs.date_histogram);
+      expect(sorted.shift()!.id!).toBe(aggs.percentiles.id + '.' + 1);
+      expect(sorted.shift()!.id!).toBe(aggs.percentiles.id + '.' + 2);
+      expect(sorted.shift()!.id!).toBe(aggs.percentiles.id + '.' + 3);
+      expect(sorted).toHaveLength(0);
+    });
+  });
+
+  describe('#toDsl', () => {
+    const schemas = new Schemas([
+      {
+        group: AggGroupNames.Buckets,
+        name: 'segment',
+      },
+      {
+        group: AggGroupNames.Buckets,
+        name: 'split',
+      },
+    ]);
+
+    beforeEach(() => {
+      mockDataServices();
+      indexPattern = stubIndexPattern as IndexPattern;
+      indexPattern.fields.getByName = name => (name as unknown) as IndexPatternField;
+    });
+
+    it('uses the sorted aggs', () => {
+      const configStates = [{ enabled: true, type: 'avg', params: { field: 'bytes' } }];
+      const ac = new AggConfigs(indexPattern, configStates, { typesRegistry });
+      const spy = jest.spyOn(AggConfigs.prototype, 'getRequestAggs');
+      ac.toDsl();
+      expect(spy).toHaveBeenCalledTimes(1);
+      spy.mockRestore();
+    });
+
+    it('calls aggConfig#toDsl() on each aggConfig and compiles the nested output', () => {
+      const configStates = [
+        { enabled: true, type: 'date_histogram', params: {}, schema: 'segment' },
+        { enabled: true, type: 'terms', params: {}, schema: 'split' },
+        { enabled: true, type: 'count', params: {} },
+      ];
+
+      const ac = new AggConfigs(indexPattern, configStates, {
+        typesRegistry,
+        schemas: schemas.all,
+      });
```

Comment:
@ppisljar I looked at the legacy browser tests, and this was how the AggConfigs was created:
```js
      const vis = new visualizationsStart.Vis(indexPattern, {
        type: 'histogram',
        aggs: [
          { type: 'date_histogram', schema: 'segment' },
          { type: 'filters', schema: 'split' },
        ],
      });

      const ac = vis.aggs;
```

However, when calling `new AggConfigs` directly as I've done here, this test only passes if I add a configState of type `count` at the end (line 317). When retrieving aggs via `Vis`, the `count` is already present.

Just want to double check that this is the expected behavior, and that I'm not artificially making this test pass.

Reply:
thats the expected behaviour, vis type has schemas defined with them and we make sure that when we init aggs it matches the schemas. histogram has a metric schema which is count by default. this test should exist on vis rather on aggconfigs, or we should be passing schemas in to aggconfigs constructor.